include/boost/gil/image_processing/fast_feature_detector.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef BOOST_GIL_IMAGE_PROCESSING_FAST_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_FAST_HPP
+
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/locator.hpp>
+#include <boost/gil/point.hpp>
+#include <algorithm>
+#include <cmath>
+#include <vector>
+namespace boost { namespace gil {
+namespace detail {
+template <typename srcview>
+bool fast_feature_detector(const srcview& buffer, int r, int c, std::vector<point_t>& points, int t)
+{
+    int valid_points_count = 16;
+    auto src_loc = buffer.xy_at(c, r);
+    std::vector<int> threshold_indicator, intensity_array(16);
+    std::vector<decltype(src_loc.cache_location(0, -1))> pointers(valid_points_count);
+    //stroring intensities of pixels on circumference beforehand to decrease runtime
+    for (int i = 0; i < 16; i++)
+    {
+        pointers[i] = src_loc.cache_location(points[i][0], points[i][1]);
+        intensity_array[i] = src_loc[pointers[i]];
+    }
+    //calculating the flags to be used during segment test
+    auto const I_p = buffer(point_t(c, r));
+    //int low,high;
+    std::transform(
+        intensity_array.begin(),
+        intensity_array.end(),
+        back_inserter(threshold_indicator),
+        [low = I_p - t, hi = I_p + t](auto const& intensity) {
+            if (intensity < low)
+                return -1;
+            else if (intensity > hi)
+                return 1;
+            else
+                return 0;
+        });
+    std::transform(
+        intensity_array.begin(),
+        intensity_array.end(),
+        back_inserter(threshold_indicator),
+        [low = I_p - t, hi = I_p + t](auto const& intensity) {
+            if (intensity < low)
+                return -1;
+            else if (intensity > hi)
+                return 1;
+            else
+                return 0;
+        });
+    //high speed test for eliminating non-corners
+    for (int i = 0; i <= 6; i += 2)
+    {
+        if (threshold_indicator[i] == 0 && threshold_indicator[i + 8] == 0)
+            return false;
+    }
+    for (int i = 1; i <= 7; i += 2)
+    {
+        if (threshold_indicator[i] == 0 && threshold_indicator[i + 8] == 0)
+            return false;
+    }
+    //final segment test
+    bool is_feature_point =
+        threshold_indicator.end() !=
+            std::search_n(threshold_indicator.begin(), threshold_indicator.end(), 9, -1) ||
+        threshold_indicator.end() !=
+            std::search_n(threshold_indicator.begin(), threshold_indicator.end(), 9, 1);
+    return is_feature_point;
+}
+template <typename srcview>
+std::ptrdiff_t
+    calculate_score(srcview& src, int i, int j, std::vector<point_t>& points, int threshold)
+{
```

Comment:
Is this formatting with indented `    calculate_score` by the `clang-format`? /cc @lpranam

Reply:
@lpranam  I will answer myself, no, it is not `clang-format` output.

Your `.clang-format` generates expected output:

```
template <typename SrcView>
std::size_t calculate_score(
    SrcView const& src,
    std::size_t i,
    std::size_t j,
    std::vector<point_t>& points,
    std::size_t threshold)
{
````

Reply:
I used the .clang-format file given by @lpranam  in the above links. But i was able to install clang-13 and not 12. May be that's the reason. Going to look into this matter again and update. Thanks for verifying @mloskot .

Reply:
I followed the comments from the PR #596 . So is this due to the .clang-format file itself? If you can kindly clarify..

Reply:
I'd suggest to use clang-format 12 but
I'd also expect clang-format 13 to give output close to clang-format 12

Reply:
I was initially unable to get clang-format 12 but now I have been able to do so.